import java.util.*;
public class roman_to_integer{
    public static void main(String[] args) {

        String s = "MCMXCIV";

        Map<Character, Integer> map = new HashMap<>();
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);

        int result = 0;

        for (int i = 0; i < s.length(); i++) {
            int curr = map.get(s.charAt(i));

            int next = 0;
            if (i + 1 < s.length()) {
                next = map.get(s.charAt(i + 1));
            }

            if (curr < next) {
                result -= curr;
            } else {
                result += curr;
            }
        }

        System.out.println(result);
    }
}

